src/sysext/sysext.c

```diff
@@ -1347,13 +1358,22 @@ static int write_extensions_file(ImageClass image_class, char **extensions, cons
         if (r < 0)
                 return log_error_errno(r, "Failed to write extension meta file '%s': %m", f);
 
+        atfd = open(f, O_CLOEXEC);
+        if (atfd < 0)
+                return log_error_errno(atfd, "Failed to open '%s': %m", f);
+
+        r = mac_selinux_fix_full(atfd, NULL, hierarchy, 0);
+        if (r < 0)
+                return log_error_errno(r, "Failed to fix SELinux label for '%s': %m", f);
```

Comment:
i really don't grok why you are labelling the hierarchy at so many places?

Reply:
mac_selinux_fix_full() only labels a file, not a directory recursive.
Or is there a function which does this recursive? I haven't found one.
If you don't label all new directories and files, they will have wrong labels, at least in all of my test cases. I tried it first with less places.

Reply:
but this only ever labels the hiararchy dir, nothing else?

files that are created fresh should be using mac_selinux_create_file_prepare_at() so that they are created correctly right-away, instead creating them under the wrong label first, and then fixing them.

Reply:
This is dealing exclusively with top level directories, let me know if there's a better way to accomplish this.